src/components/Button/ButtonContent/index.test.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react'
+import { ButtonContent, layoutType } from '.'
+import renderWithTheme from '../../../testing/renderWithTheme'
+import mentor from '../../../assets/mentor.svg'
+
+test('renders button content with layout center', () => {
+	const { baseElement } = renderWithTheme(
+		<ButtonContent layout={layoutType.center} />
+	)
+
+	expect(baseElement).toMatchInlineSnapshot(`
+		<body>
+		  <div>
+		    <div
+		      class="sc-hKgILt eNcJhZ"
+		    />
+		  </div>
+		</body>
+	`)
+})
+
+test('renders button content with layout line', () => {
+	const { baseElement } = renderWithTheme(
+		<ButtonContent layout={layoutType.line} />
+	)
+
+	expect(baseElement).toMatchInlineSnapshot(`
+		<body>
+		  <div>
+		    <div
+		      class="sc-eCssSg eOWPJF"
+		    />
+		  </div>
+		</body>
+	`)
+})
+
+// @NOTE: We don't know how to test the image
+test('renders button content with layout center and alt', () => {
+	const { getByAltText } = renderWithTheme(
+		<ButtonContent layout={layoutType.center} icon={mentor} alt="Mentor" />
+	)
+
+	const imageElement = getByAltText('Mentor')
+	expect(imageElement.tagName).toBe('IMG')
+})
+
+test('renders button content with layout center and title', () => {
+	const { getByText } = renderWithTheme(
+		<ButtonContent layout={layoutType.center} title="Mentor" />
+	)
+
+	const spanElement = getByText('Mentor')
+	expect(spanElement.tagName).toBe('SPAN')
+})
+
+test('renders button content with layout center and subtitle', () => {
+	const { getByText } = renderWithTheme(
+		<ButtonContent
+			layout={layoutType.center}
+			subtitle="Poznej nové přátele."
+		/>
+	)
+
+	const pElement = getByText('Poznej nové přátele.')
+	expect(pElement.tagName).toBe('P')
+})
+
+test('renders button content with layout center, icon, alt, title and subtitle', () => {
```

Comment:
Tenhle test bych nedělala. Budeme přece mít spoustu různých buttonů a nevím, jestli všechny budou mít nějakou ikonku nebo jestli všechny budou nutně mít podtitulek. :thinking:

Reply:
Souhlasím s tím, že budeme mít spoustu buttonů a každý bude obsahovat něco jiného...  Ale já myslela, že se mají testovat různé scénáře použití a jejich správné vypsání. Upřímně testy mi dali dost zabrat a netuším, jak mají správně vypadat. Ocenila bych pak společnou diskuzi.